src/Table/Table.tsx

```diff
@@ -0,0 +1,977 @@
+            } else {
+                result = (firstItem[columnName] - secondItem[columnName]);
+            }
+        } else {
+            if (isNaN(secondItem[columnName]) && isNaN(firstItem[columnName])) {
+                result = String(secondItem[columnName]).localeCompare(String(firstItem[columnName]), ["sw", "en"], { sensitivity: "base", ignorePunctuation: true });
+            } else {
+                result = (secondItem[columnName] - firstItem[columnName]);
+            }
+        }
+        return result;
+
+    });
+    return sortedItems;
+}
+
+/**
+ * search text in array of table row
+ * @param items the array of table rows
+ * @param keyword The keyword to search in the array
+ * @param searchFields the target field to search
+ */
+function searchTextInArray(items: Array<TableRow>, keyword: string, searchFields: Array<string>): Array<TableRow> {
+    return [...items].filter((row: TableRow) => {
+        if (keyword.trim().length === 0 || searchFields.length === 0) {
+            return true;
+        }
+        return (
+            searchFields.some((searchColumn: string) => {
+                let result: boolean = false;
+                const searchField: string = searchColumn;
+                const regEx = new RegExp(keyword, "gi");
+                if (row[searchField] === null || row[searchField] === undefined) {
+                    result = false;
+                } else if (typeof row[searchField] === "string") {
+                    result = row[searchField].search(regEx) > -1;
+                } else if (typeof row[searchField] === "number") {
+                    result = String(row[searchField]).search(regEx) !== -1;
+                }
+                return result;
+            })
+        );
+    });
+}
+
+interface ActionColumnProps {
+    actionLinks?: Array<ActionLinkItem>;
+    onActionDropped?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+    primaryActionButton?: PrimaryActionButton;
+    selectedRow: TableRow;
+    tableRef: React.RefObject<HTMLTableElement>;
+}
+
+const ActionColumn: React.FunctionComponent<ActionColumnProps> = (props: ActionColumnProps) => {
+    const btnPrimaryRandomIds = randomId("btn");
+    const [dropup, setDropup] = React.useState<boolean>(false);
+    const actionRef: React.RefObject<HTMLDivElement> = React.createRef<HTMLDivElement>();
+
+    function getActionColumnClasses() {
+        let className: string = "dropdown-content";
+        if (props.selectedRow?.actionsDropdownDropped) {
+            className += " active";
+        }
+
+        if (dropup) {
+            className += " dropup";
+        }
+
+        return className;
+    }
+
+    return (
+        <div className="action-column">
+            {props.primaryActionButton &&
+                <button
+                    id={btnPrimaryRandomIds}
```

Comment:
This shouldn't random, should be possible to pass the id as a property, setting it as random may create tracking issues

Reply:
No it cannot be tracked and thus random fit it. The actions are replicated across rows and hence allowing user to pass the ids will result in duplications.

Reply:
My Point is you are using a random function to generate a random number and it makes it hard to predict which button is which id because it looks like this **id=btn1579229636563.0505**, the ActionColumn component is called inside props.rows.map so you can use the index parameter to define a unique id for each button.

This way tracking for the first or last element of the table would be possible.

Same goes for subrows.

Reply:
My point is tracking isn't a primary hear @omarVodiaK hence the use of the random. If we want to generate unique keys in a situation where the identity of the subject is not needed, i think the random method gives more uniqueness to the id.

Reply:
you are right it gives a uniqueness but you cant be sure if it will be always unique when you use random, it might be a case even if its and edge case where randomId function will return the same result for two elements, especially if the table contains many rows.

Also the current id is useless because:

- no one can predict what is its value (getElementbyId is not possible)
-  it doesnt help for accessibility (no aria-*** attributes defined)

So my suggestion is either remove that part, or use a simple one so that at least developers can use getElementbyId